packages/client/hmi-client/src/page/project/components/tera-save-model-modal.vue

```diff
@@ -0,0 +1,148 @@
+			<template #header>
+				<h4>{{ title }}</h4>
+			</template>
+			<template #default>
+				<form @submit.prevent>
+					<label for="new-model">What would you like to call it?</label>
+					<InputText
+						id="new-model"
+						type="text"
+						v-model="newModelName"
+						placeholder="Enter a unique name for your model"
+					/>
+				</form>
+			</template>
+			<template #footer>
+				<Button label="Save" size="large" @click="save" />
+				<Button label="Cancel" class="p-button-secondary" size="large" outlined @click="cancel" />
+			</template>
+		</tera-modal>
+	</Teleport>
+</template>
+
+<script setup lang="ts">
+import { cloneDeep } from 'lodash';
+import { ref, PropType } from 'vue';
+import TeraModal from '@/components/widgets/tera-modal.vue';
+import Button from 'primevue/button';
+import InputText from 'primevue/inputtext';
+import router from '@/router';
+import { RouteName } from '@/router/routes';
+import { AssetType } from '@/types/Types';
+import type { Model } from '@/types/Types';
+import {
+	createModel,
+	// getModel,
+	updateModel
+} from '@/services/model';
+import { useProjects } from '@/composables/project';
+import { newAMR } from '@/model-representation/petrinet/petrinet-service';
+import { logger } from '@/utils/logger';
+
+const props = defineProps({
+	title: {
+		type: String,
+		default: 'Save as a new model'
+	},
+	isVisible: {
+		type: Boolean,
+		default: false
+	},
+	model: {
+		type: Object as PropType<Model>,
+		default: newAMR()
+	},
+	openOnSave: {
+		type: Boolean,
+		default: false
+	},
+	isUpdatingName: {
+		type: Boolean,
+		default: false
+	}
+});
+
+const emit = defineEmits(['close-modal', 'on-save']);
+
+const newModelName = ref<string>('');
+
+const projectResource = useProjects();
+
+// TODO: Consider letting the user just know if what they are typing currently is a duplicate (do not prevent them from saving)
+// const isValidName = ref<boolean>(true);
+// const invalidInputStyle = computed(() => (!isValidName.value ? 'p-invalid' : ''));	v-bind:class="invalidInputStyle"
+
+async function saveAs() {
+	const newModel = cloneDeep(props.model);
+	newModel.header.name = newModelName.value.trim();
+
+	const modelData = await createModel(newModel);
+	if (!modelData) return;
+
+	const projectId = projectResource.activeProject.value?.id;
+	await projectResource.addAsset(AssetType.Model, modelData.id, projectId);
+
+	logger.info(
+		`${modelData.name} saved successfully in project ${projectResource.activeProject.value?.name}.`
+	);
+
+	emit('on-save', modelData);
+
+	if (props.openOnSave) {
+		router.push({
+			name: RouteName.Project,
+			params: {
+				pageType: AssetType.Model,
+				assetId: modelData.id
+			}
+		});
+	}
+
+	emit('close-modal');
+}
+
+async function updateName() {
+	const modelToUpdate = cloneDeep(props.model);
+	modelToUpdate.header.name = newModelName.value.trim();
+
+	const response = await updateModel(modelToUpdate);
+	if (!response) return;
+
+	await projectResource.refresh();
```

Comment:
This doesn't need to be `await`, it can happen on the background that's okay.
Even better will be to call this method in the `updateModel` call

Reply:
I left a TODO comment, I'll do the project refresh when the model supports all assets